common/determined_common/api/experiment.py

```diff
@@ -25,48 +26,48 @@ def activate_experiment(master_url: str, exp_id: int) -> None:
 
 @authentication_required
 def logs(args: Namespace) -> None:
-    last_offset, last_state = 0, None
-
-    def print_logs(offset: Optional[int], limit: Optional[int] = 5000) -> Any:
-        nonlocal last_offset, last_state
-        path = "trials/{}/logsv2?".format(args.trial_id)
+    def print_logs(
+        offset: Optional[int], limit: Optional[int] = 5000, follow: bool = False
+    ) -> None:
+        path = "/api/v1/trials/{}/logs?".format(args.trial_id)
         if offset is not None:
             path += "&offset={}".format(offset)
         if limit is not None:
             path += "&limit={}".format(limit)
-        logs = api.get(args.master, path).json()
-        for log in logs:
-            print(log["message"], end="")
-            last_state = log["state"]
-        return logs[-1]["id"] if logs else last_offset
+        if follow:
+            path += "&follow=true"
+        for f in [
+            "agent_id",
+            "container_id",
+            "source",
+            "std_type",
+            "level",
+            "timestamp_before",
+            "timestamp_after",
+        ]:
+            if getattr(args, f) is not None:
+                for v in getattr(args, f):
+                    path += "&{}={}".format(f, v)
```

Comment:
blocking: Escape the values with [`urllib.parse.quote_plus`](https://docs.python.org/3/library/urllib.parse.html#urllib.parse.quote_plus) or else `&` will lead to confusion.

non-blocking: I also wouldn't mind moving this whole thing over to using [`urllib.parse.urlencode`](https://docs.python.org/3/library/urllib.parse.html#urllib.parse.urlencode) (`doseq=True` will be required).